libd/src/d/semantic/symbol.d

```diff
@@ -732,11 +743,81 @@ struct SymbolAnalyzer {
 			
 			baseMethods = baseMethods[0 .. i];
 		}
-		
+
 		c.members = cast(Symbol[]) baseFields;
 		c.members ~= baseMethods;
 		scheduler.require(members);
 		c.members ~= members;
+
+		bool found = false;
+		foreach (i; c.interfaces){
+			scheduler.require(i);
+
+			auto vtblType = Type.get(BuiltinType.Void).getPointer(TypeQualifier.Immutable);
+			
+			// XXX: d is hijacked without explicit import
+			import d.context.name : BuiltinName;
+			
+			// TODO: use defaultinit.
+			auto ivtbl = new Field(d.location, 0, vtblType, BuiltinName!"__ivtbl", null);
+			c.members ~= ivtbl;
+			//c.ivtblOffset[i]
+			// TODO: uint[Interface]
+			MemberLoop: foreach (m; i.members) {
+				if (auto interfaceMethod = cast(Method) m) {
+					auto mt = interfaceMethod.type;
+					auto rt = mt.returnType;
+					auto ats = mt.parameters[1 .. $];
+					import std.stdio;
+					found = false;
+					ClassLoop: foreach (member;  c.members) {
+						if (auto classMethod = cast(Method) member) {
+							writeln("cl: 1");
+							if(classMethod.name != interfaceMethod.name)
+								continue;
+							writeln("cl: 2");
+							auto ct = classMethod.type;
+							if(ct.isVariadic != mt.isVariadic)
+								continue;
+							writeln("cl: 3");
+							auto crt = ct.returnType;
+							auto cts = ct.parameters[1 .. $];
+							if (ats.length != cts.length || rt.isRef != crt.isRef) {
+								continue;
+							}
+							writeln("cl: 4");
+							if (implicitCastFrom(pass, rt.getType(), crt.getType()) < CastKind.Exact) {
+								continue;
+							}
+							writeln("cl: 5");
+							import std.range;
+							foreach(at, ct; lockstep(ats, cts)) {
+								if (at.isRef != ct.isRef) {
+									continue ClassLoop;
+								}
+								
+								if (implicitCastFrom(pass, ct.getType(), at.getType()) < CastKind.Exact) {
+									continue ClassLoop;
+								}
+							}
+							writeln("cl: 6");
+							// found method in class
+							found = true;
+							break;
+						}
```

Comment:
There is massive duplication here with matching method of the base class. Code duplication hurt maintainability.